shelley/chain-and-ledger/executable-spec/src/Cardano/Ledger/Val.hs

```diff
@@ -88,6 +89,10 @@ sumVal xs = foldl (<+>) mempty xs
 invert :: Val t => t -> t
 invert x = (-1 :: Integer) <×> x
 
+-- returns a Value containing only the coin (ada) tokens from the input Value
+adaOnly :: Val v => v -> v
```

Comment:
maybe this should return Bool (ie do the check)?

Reply:
good point!